diff --git a/slippers/props.py b/slippers/props.py
index e1aa4dc..4afb132 100644
--- a/slippers/props.py
+++ b/slippers/props.py
@@ -27,13 +27,13 @@ class Props(Mapping):
         types: Dict[str, type],
         defaults: Dict[str, Any],
     ):
-        self.attributes = attributes
+        self._attributes = {**attributes}
         self.types = types
         self.defaults = defaults
 
     def __getitem__(self, key: str) -> Any:
-        if key in self.attributes:
-            return self.attributes[key]
+        if key in self._attributes:
+            return self._attributes[key]
 
         if key in self.defaults:
             return self.defaults[key]
@@ -41,13 +41,13 @@ class Props(Mapping):
         return None
 
     def __setitem__(self, key: str, value: Any) -> None:
-        self.attributes[key] = value
+        self._attributes[key] = value
 
     def __iter__(self):
-        return iter({**self.attributes, **self.defaults})
+        return iter({**self._attributes, **self.defaults})
 
     def __len__(self):
-        return len({**self.attributes, **self.defaults})
+        return len({**self._attributes, **self.defaults})
 
     @classmethod
     def from_string(cls, attributes: Dict[str, Any], code: str) -> "Props":
@@ -71,18 +71,23 @@ class PropError:
     actual: Optional[type]
 
 
-def check_prop_types(*, props: Props):
+def check_prop_types(
+    *,
+    attributes: Dict[str, Any],
+    types: Dict[str, type],
+    defaults: Dict[str, Any],
+) -> List[PropError]:
     """Check that props are of the correct type"""
 
     errors = []
 
     # Check for missing props
-    for name, expected in props.types.items():
-        if name not in props.attributes:
+    for name, expected in types.items():
+        if name not in attributes:
             if get_origin(expected) is Union and type(None) in get_args(expected):
                 # Props with Optional types are not required
                 continue
-            elif name in props.defaults:
+            elif name in defaults:
                 # Props with defaults are not required
                 continue
             else:
@@ -97,9 +102,9 @@ def check_prop_types(*, props: Props):
                 )
 
     # Check for invalid props
-    for name, actual in props.attributes.items():
-        if name in props.types:
-            expected = props.types[name]
+    for name, actual in attributes.items():
+        if name in types:
+            expected = types[name]
             try:
                 check_type(name, actual, expected)
             except TypeError:
@@ -113,8 +118,8 @@ def check_prop_types(*, props: Props):
                 )
 
     # Check for extra props
-    for name, actual in props.attributes.items():
-        if name not in props.types:
+    for name, actual in attributes.items():
+        if name not in types:
             errors.append(
                 PropError(
                     error="extra",
diff --git a/slippers/templatetags/slippers.py b/slippers/templatetags/slippers.py
index 34c2c35..1e568f2 100644
--- a/slippers/templatetags/slippers.py
+++ b/slippers/templatetags/slippers.py
@@ -110,7 +110,11 @@ class ComponentNode(template.Node):
             props = Props.from_string(attributes, source_front_matter)
 
             if settings.SLIPPERS_RUNTIME_TYPE_CHECKING:
-                prop_errors = check_prop_types(props=props)
+                prop_errors = check_prop_types(
+                    attributes=attributes,
+                    types=props.types,
+                    defaults=props.defaults,
+                )
 
             if "shell" in settings.SLIPPERS_TYPE_CHECKING_OUTPUT and prop_errors:
                 print_errors(
